lib/surface/compiler.ex

```diff
@@ -262,7 +265,9 @@ defmodule Surface.Compiler do
          meta: meta
        }}
     else
-      _ -> {:error, {"failed to parse slot", meta.line}, meta}
+      _ ->
+        short_slot_syntax? = not has_attribute?(attributes, "name")
+        missing_slot(meta.caller.module, name, meta, defined_slot_names, short_slot_syntax?)
```

Comment:
Maybe we should call this function `raise_missing_slot` or even `raise_missing_slot_error` to make it clear what they do. The same goes to all similar functions. WDYT?

Reply:
since it's going to raise, should we also append a `!` to the name?